### DISCUSSION TITLE
compile time error - any ideas?

### DISCUSSION THREAD

**Discussion by Pilot001** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#459] [Q&A] [answered]
During compilation of v3.5.1 in Arduino I get the follow error message:
```
bestway.ino: In function 'void handleESPInfo()':
bestway:2559:31: error: cannot pass objects of non-trivially-copyable type 'class String' through '...'
     ESP.getMaxFreeBlockSize() );
                               ^
```
Any ideas how to fix that?
Thank you in advance!
And BTW: Thanks for this great project!

**Comment by visualapproach** [comment]
Seems you have a parenthesis too much, but I don't know. You are free to use any IDE but I can only support VCS + platformio.
Thx

**Comment by Pilot001** [comment]
I understand, but this is not an issue of the IDE, as the error message comes from the compiler. And this is the original code as I did not do any changes to the bestway.cpp file (except for renaming it to bestway.ino).

**Comment by visualapproach** [comment]
There is no file named 'bestway.cpp' that I can recall. Maybe you meant 'main.cpp' but still, it works in platformio and I don't have the time to support anything else.

**Comment by Pilot001** [comment, marked as answer]
I'm sorry, of course "main.cpp"...
It took me a while to find the culprit, as the point the compiler states as wrong is just the end of the whole definition...
The problem is the ESP.getCoreVersion() call, which returns a String object, but printf (or snprintf) with %s officially only supports char*, so my suggestion is this diff:
```
--- bestway.ino.org 2022-12-27 22:41:15.000000000 +0100
+++ bestway.ino     2023-03-04 22:31:22.813588548 +0100
@@ -2552,7 +2552,7 @@
   snprintf(response, BUFSIZE, response_template,
     stacksize,
     ESP.getFreeHeap(),
-    ESP.getCoreVersion(),
+    &ESP.getCoreVersion()[0],
     ESP.getCpuFreqMHz(),
     ESP.getCycleCount(),
     ESP.getFreeContStack(),
```

**Reply by visualapproach** [reply]
Thanks, I fixed it in dev-branch, although I used .c_str() to be consistent with the rest of the code.
